src/hello_freertos.c

```diff
@@ -1,54 +1,213 @@
+    }
+}
+
+// 4. Disable compiler optimizations with a tight loop
+#pragma GCC optimize ("O0") // Disable compiler optimizations for this function
+void blink_task_no_optimize(__unused void *params) {
+    gpio_init(OUT_PIN);            // Initialize the GPIO pin
+    gpio_set_dir(OUT_PIN, GPIO_OUT); // Set it as an output pin
+
+    while (true) {
+        gpio_put(OUT_PIN, true);    // Turn on LED
+
+        // Tight loop for CPU usage (no delay)
+        for (int i = 0; i < 10000000; i++) {
+            // Perform some trivial but CPU-heavy operations (e.g., multiplication)
+            volatile int dummy = i * 3;  // Keeps CPU busy with computation
+        }
+
+        gpio_put(OUT_PIN, false);   // Turn off LED
+
+        // Tight loop for CPU usage (no delay)
+        for (int i = 0; i < 10000000; i++) {
+            // Another CPU-heavy operation
+            volatile int dummy = i * 2;  // Keeps CPU busy with computation
+        }
+    }
+}
+
+// 5. Run a tight loop to trigger high CPU activity
+void blink_task_tight_loop(__unused void *params) {
+    gpio_init(OUT_PIN);            // Initialize the GPIO pin
+    gpio_set_dir(OUT_PIN, GPIO_OUT); // Set it as an output pin
+
+    volatile int result = 0;
+    while (true) {
+        for (volatile int i = 1; i <= 10000; i++) {
+            result += i * i; // Perform multiply and accumulate
+        }
+        gpio_put(OUT_PIN, !gpio_get(OUT_PIN)); // Toggle LED
     }
 }
 
-int main( void )
-{
+int main(void) {
+    // Initialize standard I/O for printing messages
     stdio_init_all();
-    const char *rtos_name;
-    rtos_name = "FreeRTOS";
-    TaskHandle_t task;
-    xTaskCreate(main_task, "MainThread",
-                MAIN_TASK_STACK_SIZE, NULL, MAIN_TASK_PRIORITY, &task);
+
+    // Uncomment the desired task to run ---------
+
+    //Scenario 1
+    // xTaskCreate(blink_task, "BlinkThread",
+    //             BLINK_TASK_STACK_SIZE, NULL, BLINK_TASK_PRIORITY, NULL);
+
+    //Scenario 2            
+    // xTaskCreate(blink_task_thread, "BlinkThread",
+    //             BLINK_TASK_STACK_SIZE, NULL, BLINK_TASK_PRIORITY, NULL);
+    // xTaskCreate(blink_task_thread2, "BlinkThread2",
+    //             BLINK_TASK_STACK_SIZE, NULL, BLINK_TASK_PRIORITY2, NULL);
+
+    //Scenario 3   
+    // xTaskCreate(blink_task_busy_loop, "BlinkThread",
+    //             BLINK_TASK_STACK_SIZE, NULL, BLINK_TASK_PRIORITY, NULL);
+
+    //Scenario 4
+    // xTaskCreate(blink_task_no_optimize, "BlinkThread",
+    //             BLINK_TASK_STACK_SIZE, NULL, BLINK_TASK_PRIORITY, NULL);
+
+    //Scenario 5
+    xTaskCreate(blink_task_tight_loop, "BlinkThread",
+                BLINK_TASK_STACK_SIZE, NULL, BLINK_TASK_PRIORITY, NULL);
+
+    // Start the FreeRTOS scheduler
     vTaskStartScheduler();
-    return 0;
+
+    return 0; // Should never reach here unless scheduler fails
 }
+
+// ----- Original Code to debug that was the first pushed to work_branch
```

Comment:
I would move this to a debug branch to keep the work branch cleaner. In that case there is also no need to have both sections of code in either branch.